src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -103,6 +103,36 @@ protected CompareExchangeCommandBase(string database, string key, long index, Js
             }
         }
 
+        public static unsafe void GetDbPrefixAndLastSlices(
+            ByteStringContext allocator, string db,
+            out (Slice Slice, ByteStringContext.ExternalScope Scope) prefix,
+            out (Slice Slice, ByteStringContext.ExternalScope Scope) last)
+        {
+            var reservedSpace = Encoding.UTF8.GetMaxByteCount(db.Length) + 1;
+
+            allocator.Allocate(reservedSpace, out var prefixBuffer); // db + '/'
+            allocator.Allocate(reservedSpace, out var lastBuffer); // db + (char)('/'+1)
+
+            fixed (char* pDb = db)
+            {
+                var len = Encoding.UTF8.GetBytes(pDb, db.Length, prefixBuffer.Ptr, prefixBuffer.Length);
+
+                prefixBuffer.Ptr[len] = (byte)'/';
+                prefixBuffer.Truncate(len + 1);
+
+                allocator.ToLowerCase(ref prefixBuffer);
+
+                prefixBuffer.CopyTo(lastBuffer.Ptr);
+
+                lastBuffer.Ptr[len++] = '/' + 1;
```

Comment:
We pass it to `SeekOneBackwardFrom` method. In order to make sure it works as expected we need tests which have more than one database and multiple compare exchange values for each. This way we'll know that we get correct values as expected.